src/panoptes/pocs/focuser/astromechanics.py

```diff
@@ -0,0 +1,263 @@
+import io
+import serial
+from warnings import warn
+from contextlib import suppress
+
+from panoptes.pocs.focuser import AbstractFocuser
+
+
+class Focuser(AbstractFocuser):
+    """
+    Focuser class for control of a Canon DSLR lens via an Astromechanics Engineering Canon EF/EF-S adapter.
+
+    Args:
+        name (str, optional): default 'Astromechanics Focuser'
+        model (str, optional): default 'Canon EF/EF-S'
+        initial_position (int, optional): if given the focuser will drive to this encoder position
+            following initialisation.
+        dev_node_pattern (str, optional): Unix shell pattern to use to identify device nodes that
+            may have a focuser adaptor attached. Default is '/dev/tty.usbserial-00*?'
+
+    Additional positonal and keyword arguments are passed to the base class, AbstractFocuser. See
+    that class' documentation for a complete list.
+    """
+
+    # Class variable to cache the device node scanning results
+    _adaptor_nodes = None
+
+    # Class variable to store the device nodes already in use. Prevents scanning known focuser adaptors and
+    # acts as a check against those ones assigned to incorrect ports.
+    _assigned_nodes = []
+
+    def __init__(self,
+                 name='Astromechanics Focuser Controller',
+                 model='astromechanics',
+                 initial_position=None,
+                 dev_node_pattern='/dev/tty.usbserial-00*',
```

Comment:
Neither the `initial_position` or the `dev_node_pattern` are actually used anywhere.